### PR TITLE
Generate msi for win-arm64 installer

### LINKED ISSUE #20672: Release Windows on Arm64 MSIs/packages
### Summary of the new feature / enhancement

Please consider building and releasing official Windows on Arm64 releases with easy to use MSIs for PowerShell. Windows on Arm64 has been supported for a while now and currently the only way to install it on Arm64 machines is through the zip package which is a manual process and not ideal. Windows Store releases are also not the way to go here (at least exclusively) because PowerShell has Store version-only bugs.

Due to the lack of MSI packages, many users may end up inadvertently with the x64 version which offers a subpar experience in performance. Further, scripts (like those that install other programs) running in the x64 environment may read system variables under emulation incorrectly and assume they're on an x64 system, further exacerbating the problem.

### Proposed technical implementation details (optional)

Package and release MSI installer packages for Windows on Arm64 for all future releases.

### PR DESCRIPTION
**Pull request by anamnavi**
# PR Summary
Generate msi for arm powershell win-arm64 package instead of msixbundle. This requires us to use WixToolset 3.14.

Resolves #20672 

## PR Checklist

- [x] [PR has a meaningful title](https://github.com/PowerShell/PowerShell/blob/master/.github/CONTRIBUTING.md#pull-request---submission)
    - Use the present tense and imperative mood when describing your changes
- [x] [Summarized changes](https://github.com/PowerShell/PowerShell/blob/master/.github/CONTRIBUTING.md#pull-request---submission)
- [x] [Make sure all `.h`, `.cpp`, `.cs`, `.ps1` and `.psm1` files have the correct copyright header](https://github.com/PowerShell/PowerShell/blob/master/.github/CONTRIBUTING.md#pull-request---submission)
- [ ] This PR is ready to merge and is not [Work in Progress](https://github.com/PowerShell/PowerShell/blob/master/.github/CONTRIBUTING.md#pull-request---work-in-progress).
    - If the PR is work in progress, please add the prefix `WIP:` or `[ WIP ]` to the beginning of the title (the `WIP` bot will keep its status check at `Pending` while the prefix is present) and remove the prefix when the PR is ready.
- **[Breaking changes](https://github.com/PowerShell/PowerShell/blob/master/.github/CONTRIBUTING.md#making-breaking-changes)**
    - [ ] None
    - **OR**
    - [ ] [Experimental feature(s) needed](https://github.com/MicrosoftDocs/PowerShell-Docs/blob/main/reference/7.3/Microsoft.PowerShell.Core/About/about_Experimental_Features.md)
        - [ ] Experimental feature name(s): 
- **User-facing changes**
    - [ ] Not Applicable
    - **OR**
    - [ ] [Documentation needed](https://github.com/PowerShell/PowerShell/blob/master/.github/CONTRIBUTING.md#pull-request---submission)
        - [ ] Issue filed: 
- **Testing - New and feature**
    - [ ] N/A or can only be tested interactively
    - **OR**
    - [ ] [Make sure you've added a new test if existing tests do not effectively test the code changed](https://github.com/PowerShell/PowerShell/blob/master/.github/CONTRIBUTING.md#before-submitting)
- **Tooling**
    - [ ] I have considered the user experience from a tooling perspective and don't believe tooling will be impacted.
    - **OR**
    - [ ] I have considered the user experience from a tooling perspective and opened an issue in the relevant tool repository. This may include:
        - [ ] Impact on [PowerShell Editor Services](https://github.com/PowerShell/PowerShellEditorServices) which is used in the [PowerShell extension](https://github.com/PowerShell/vscode-powershell) for VSCode
        (which runs in a different PS Host).
            - [ ] Issue filed: 
        - [ ] Impact on Completions (both in the console and in editors) - one of PowerShell's most powerful features.
            - [ ] Issue filed: 
        - [ ] Impact on [PSScriptAnalyzer](https://github.com/PowerShell/PSScriptAnalyzer) (which provides linting & formatting in the editor extensions).
            - [ ] Issue filed: 
        - [ ] Impact on [EditorSyntax](https://github.com/PowerShell/EditorSyntax) (which provides syntax highlighting with in VSCode, GitHub, and many other editors).
            - [ ] Issue filed:

### CODE REVIEW

File: assets/wix/bundle.wxs [hunk 1]
```diff
@@ -9,6 +9,15 @@
     <?else?>
       <?define UpgradeCode = $(var.UpgradeCodeRelease)?>
     <?endif?>
+  <?elseif $(sys.BUILDARCH) = "ARM64"?>
+    <?define ExplorerContextMenuDialogText = "&$(var.ProductName) $(var.SimpleProductVersion) ($(sys.BUILDARCH))"?>
+    <?define UpgradeCodePreview = "f064cf16-97b7-4550-b392-ce0f4a6823a3"?>
```
Comment (TravisEz13) [inline assets/wix/bundle.wxs:14]: product and bundle upgrade codes need to be unique

**Review by TravisEz13 — DISMISSED** [review]
Please address comment

**Review by TravisEz13 — COMMENTED** [review]
Looks good but we should do a test release build

**Review by adityapatwardhan — APPROVED** [review]

**Comment by anamnavi** [comment]
@adityapatwardhan confirmed that the package can be ran and installed on an Windows 11 arm machine :)

**Pull request merged**